### PR TITLE
IGN -> GZ

### PR DESCRIPTION
**Pull request by ahcorde**
Build on top of this other PR https://github.com/ArduPilot/ardupilot_gazebo/pull/26

### CODE REVIEW

**Comment by srmainwaring** [comment]
@khancyr yes that's right - Garden only.

@ahcorde I've been making similar changes for another Gazebo plugin (my marine ocean one), wondering if it is worth waiting until the Gazebo Garden interface is finalised before merging this change as I've been through a few iterations catching up as the namespace migration progresses; on the other hand the namespace changes may be breaking for Garden.

**Comment by ahcorde** [comment]
we are still migrating `ign -> gz`, 

I openned this PR just in case someone want to try this with Garden. I will try to keep it updated

**Comment by srmainwaring** [comment]
@ahcorde is the reset functionality required by the ign -> gz migration or is it optional? If it's optional we should split that into a separate PR.

**Comment by ahcorde** [comment]
Hi @srmainwaring @khancyr,

this is ready for a review

File: README.md [hunk 1]
```diff
@@ -1,35 +1,35 @@
...
+This is ArduPilot official plugin for Gazebo Simulator plugin
 It replaces the old Gazebo plugin to bring support of the next gen Gazebo simulator.
 It also brings support for more features :
 - more flexible data exchange between SITL and Ignition with JSON data sharing.
 - more sensors supported.
 - true Simulation lockstepping. It is now possible to use GDB to stop the Ignition time for debugging.
 - Better 3D rendering
 
-The project is composed of an Ignition plugin to connect to ArduPilot SITL (Software In The Loop) and some example models and worlds.
+The project is composed of an Gazebo plugin to connect to ArduPilot SITL (Software In The Loop) and some example models and worlds.
```
Comment (srmainwaring) [inline README.md:10]: minor gram. chg.  *an Gazebo* to *a Gazebo*

File: README.md [hunk 2]
```diff
@@ -1,35 +1,35 @@
...
 
 ## Disclaimer :
 The plugin is currently working, but we are working into bringing support for more feature and refine the API.
 
 ## Prerequisites :
-Ignition Fortress is supported on Ubuntu Bionic, Focal and Jammy. If you are running Ubuntu as a virtual machine you will need at least Ubuntu 20.04 (Focal) in order to have the OpenGL support required for the `ogre2` render engine.
+Gazebo Sim Garden is supported on Ubuntu Focal and Jammy. If you are running Ubuntu as a virtual machine you will need at least Ubuntu 20.04 (Focal) in order to have the OpenGL support required for the `ogre2` render engine.
 
-Follow the instructions for a [binary install of ignition fortress](https://ignitionrobotics.org/docs/fortress/install) and verify that ignition gazebo is running correctly.
+Follow the instructions for a [binary install of ignition fortress](https://ignitionrobotics.org/docs/garden/install) and verify that ignition gazebo is running correctly.
```
Comment (srmainwaring) [inline README.md:18]: Link is to garden, comment is for fortress. Suggest change comment to garden as well.

File: models/iris_with_standoffs/model.sdf [hunk 3]
```diff
@@ -538,11 +538,11 @@
     <static>0</static>
 
     <!-- plugins -->
-    <plugin filename="libignition-gazebo-joint-state-publisher-system.so"
-      name="ignition::gazebo::systems::JointStatePublisher">
+    <plugin filename="libgz-sim-joint-state-publisher-system.so"
```
Comment (srmainwaring) [inline models/iris_with_standoffs/model.sdf:541]: We can drop the `lib` prefix and `.so` suffix on the library names, so

`libgz-sim-joint-state-publisher-system.so` to `gz-sim-joint-state-publisher-system`, etc.

File: src/ArduPilotPlugin.cc [hunk 4]
```diff
@@ -1032,7 +1076,7 @@ void ignition::gazebo::systems::ArduPilotPlugin::ApplyMotorForces(
       else if (this->dataPtr->controls[i].type == "POSITION")
       {
         //TODO: figure out whether position control matters, and if so, how to use it.
-        ignwarn << "Failed to do position control on joint " << i << 
+        ignwarn << "Failed to do position control on joint " << i <<
```
Comment (srmainwaring) [inline src/ArduPilotPlugin.cc:1079]: Change `ignwarn` to `gzwarn` and similarly for `ignmsg`?

File: src/ArduPilotPlugin.cc [hunk 5]
```diff
@@ -1480,7 +1524,7 @@ void ignition::gazebo::systems::ArduPilotPlugin::SendState() const
         this->dataPtr->fcu_port_out);
 
 #if DEBUG_JSON_IO
-    igndbg << "sent " << bytes_sent <<  " bytes to " 
+    igndbg << "sent " << bytes_sent <<  " bytes to "
         << this->dataPtr->fcu_address << ":" << this->dataPtr->fcu_port_out << "\n"
```
Comment (srmainwaring) [inline src/ArduPilotPlugin.cc:1528]: Change `igndbg` to `gzdbg`.

**Review by srmainwaring — CHANGES_REQUESTED** [review]
Hi @ahcorde, thanks for `ign` -> `gz` migration changes. I've made similar changes to my local branch and they seem to be in line.

The one feature I have not checked is the reset interface - so I'll need to run your branch to check that. Otherwise it's just a few minor changes here and there - mainly using the library short form in the plugins and switching the message ostream to the `gz` version.

**Review by srmainwaring — APPROVED** [review]
@ahcorde this LGTM, thanks!

Have tested `iris_arducopter_runway.world` on latest pull of gz-sim (d90fab) against ArduPilot master on macOS Monterey Version 12.5, Xcode  13.2.1 and it's working well.

**Pull request merged**